### PR TITLE
Fix to SSRF and User controled Path expression

### PR DESCRIPTION
**Pull request by samirparhi-dev**
Fix to SSRF and User controled Path expression

### CODE REVIEW

File: src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java [hunk 1]
```diff
@@ -277,35 +304,37 @@ public String sendFoetalMonitorTestDetails(FoetalMonitor request, String auth) t
...
 				request.setDeleted(true);
 				foetalMonitorRepo.save(request);
 			}
-			if(jsnOBJ.get("status") !=null && jsnOBJ.get("message") !=null)
-//			throw new Exception("Unable to raise test request, error is : " + ("status code "+(jsnOBJ.get("status").getAsString())
-//					+","+(jsnOBJ.get("message").getAsString())));
-				throw new Exception("Unable to raise test request, error is : " + (jsnOBJ.get("message").getAsString()));
+			if (jsnOBJ.get("status") != null && jsnOBJ.get("message") != null)
+				// throw new Exception("Unable to raise test request, error is : " + ("status
+				// code "+(jsnOBJ.get("status").getAsString())
```
Comment (Anjana2310) [inline src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java:326]: Please remove the commented out code

File: src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java [hunk 2]
```diff
@@ -277,35 +304,37 @@ public String sendFoetalMonitorTestDetails(FoetalMonitor request, String auth) t
 			} else
 				throw new RuntimeException("Unable to generate foetal monitor id in TM");
 
-		} 
+		}
 		/**
 		 * @author SH20094090
 		 * @purpose To get response body in case of exception
```
Comment (Anjana2310) [inline src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java:310]: Please remove the author

File: src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java [hunk 3]
```diff
@@ -176,34 +181,54 @@ public int updateFoetalMonitorData(FoetalMonitor foetalMonitorDataOutside) throw
...
-			con.setDoInput(true);
-			filePathLocal = foetalMonitorFilePath + "/" + timeStamp.toString() + ".pdf";
-			Path path = Paths.get(filePathLocal);
-			Files.copy(con.getInputStream(), path, StandardCopyOption.REPLACE_EXISTING);
 
-			// base64 = readPDFANDGetBase64(filePathLocal);
+			// "Best Practice": Set headers as needed for the specific API
+			/*
+			 * con.addRequestProperty("User-Agent", "Your-User-Agent");
+			 * con.addRequestProperty("Authorization", "Bearer Your-AccessToken");
```
Comment (Anjana2310) [inline src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java:197]: Please remove the commented out code

**Review by Anjana2310 — COMMENTED** [review]
@samirparhi-dev Please address the comments

**Review by ravishanigarapu — COMMENTED** [review]
Please attach test scenarios screenshots

**Comment by samirparhi-dev** [comment]
> Please attach test scenarios screenshots

@ravishanigarapu : The changes are done w.r.t to transform the existing code to a secure code. I am sure you have the document which defines the functionality of each method. Please point us to the document so that we will be able to write the unit testcase for it. Else we encourage you/your team member to write the corresponding unit test for all the class and method which will enable us to validate our code changes before we create a PR.

@Anjana2310 FYI.

**Comment by samirparhi-dev** [comment]
> Please attach test scenarios screenshots

@ravishanigarapu : The changes are done w.r.t to transform the existing code to a secure code. I am sure you have the document which defines the functionality of each method. Please point us to the document so that we will be able to write the unit testcase for it. Else we encourage you to write the corresponding unit test for all the class and method which will enable us to validate our code changes before we create a PR.

@Anjana2310 FYI.

File: src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java (hunk 1 above)
Comment (samirparhi-dev) [reply src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java:326]: Sure. I will do it soon

File: src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java (hunk 3 above)
Comment (samirparhi-dev) [reply src/main/java/com/iemr/tm/service/foetalmonitor/FoetalMonitorServiceImpl.java:197]: Sure. I will do it soon

**Comment by drtechie** [comment]
@samirparhi-dev any update on this PR?

**Review by ravishanigarapu — COMMENTED** [review]
resolve conflicts

**Review by ravishanigarapu — APPROVED** [review]
looks ok

**Pull request merged**